src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -151,6 +160,16 @@ class BlockchainImpl(
 
   override def getTotalDifficultyByHash(blockhash: ByteString): Option[BigInt] = totalDifficultyStorage.get(blockhash)
 
+  override def getAccount(address: Address, blockNumber: BigInt): Option[Account] =
+    getBlockHeaderByNumber(blockNumber).flatMap { bh =>
+      val mpt = new MerklePatriciaTrie[Address, Account](
+        Some(bh.stateRoot.toArray),
```

Comment:
Minor detail: Maybe we should use the `def apply[K, V](rootHash: Array[Byte], source: NodeStorage, hashFn: HashFn): MerklePatriciaTrie[K, V]` constructor? I think this constructor should be private right?